test/unit/services/pgp.service.test.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  generateNewKeys,
+  encryptMessageWithPublicKey,
+  decryptMessageWithPrivateKey,
+} from '../../../src/app/crypto/services/pgp.service';
+
+describe('Encryption and Decryption', () => {
+  it('should generate new keys', async () => {
+    const keys = await generateNewKeys();
+
+    expect(keys).toHaveProperty('privateKeyArmored');
+    expect(keys).toHaveProperty('publicKeyArmored');
+    expect(keys).toHaveProperty('revocationCertificate');
+  });
+
+  it('should encrypt a message with the given public key', async () => {
+    const keys = await generateNewKeys();
+    const publicKeyInBase64 = keys.publicKeyArmored;
+    const message = 'This is a test message';
+
+    const encryptedMessage = await encryptMessageWithPublicKey({
+      message,
+      publicKeyInBase64,
+    });
+
+    expect(encryptedMessage).toBeDefined();
+  });
+
+  it('should generate keys, encrypt and decrypt a message successfully', async () => {
+    // Step 1: Generate new keys
+    const keys = await generateNewKeys();
+    expect(keys).toHaveProperty('privateKeyArmored');
```

Comment:
Hey, as you have asserted this on the other test case, you could remove these assertions or put that at the end of the test, so we prepare first, then act, then assert